examples/08_experimental/plot_new_surface_bids_analysis.py

```diff
@@ -0,0 +1,180 @@
+   also known as the Freesurfer template :term:`mesh`
+   downsampled to about 10k nodes per hemisphere.
+
+#. Run the first level model objects.
+
+#. Fit a second level model on the fitted first level models.
+
+Notice that in this case the preprocessed :term:`bold<BOLD>` images
+were already normalized to the same :term:`MNI` space.
+"""
+
+# %%
+# Fetch example :term:`BIDS` dataset
+# ----------------------------------
+# We download a simplified :term:`BIDS` dataset
+# made available for illustrative purposes.
+# It contains only the necessary information
+# to run a statistical analysis using Nilearn.
+# The raw data subject folders only contain bold.json and events.tsv files,
+# while the derivatives folder includes the preprocessed files preproc.nii
+# and the confounds.tsv files.
+from nilearn.datasets import fetch_language_localizer_demo_dataset
+
+data_dir, _ = fetch_language_localizer_demo_dataset()
+
+# %%
+# Here is the location of the dataset on disk.
+print(data_dir)
+
+# %%
+# Obtain automatically FirstLevelModel objects and fit arguments
+# --------------------------------------------------------------
+# From the dataset directory we automatically obtain
+# the FirstLevelModel objects
+# with their subject_id filled from the :term:`BIDS` dataset.
+# Moreover, we obtain for each model
+# a dictionary with run_imgs, events and confounder regressors
+# since in this case a confounds.tsv file is available
+# in the :term:`BIDS` dataset.
+# To get the first level models we only have to specify the dataset directory
+# and the task_label as specified in the file names.
+from nilearn.glm.first_level import first_level_from_bids
+
+task_label = "languagelocalizer"
+models, run_imgs, events, confounds = first_level_from_bids(
+    data_dir,
+    task_label,
+    img_filters=[("desc", "preproc")],
+    hrf_model="glover + derivative",
+    n_jobs=2,
+)
+
+
+# %%
+# Project :term:`fMRI` data to the surface: First get fsaverage5.
+from nilearn.experimental.surface import SurfaceImage, load_fsaverage
+
+fsaverage5 = load_fsaverage()
+
+# %%
+# The projection function simply takes the :term:`fMRI` data and the mesh.
+# Note that those correspond spatially, as they are both in :term:`MNI` space.
+from nilearn import surface
+
+# %%
+# Empty lists in which we are going to store activation values.
+z_scores_left = []
+z_scores_right = []
+for first_level_glm, fmri_img, confound, events in zip(
+    models, run_imgs, confounds, events
+):
+    texture_left = surface.vol_to_surf(
+        fmri_img[0], fsaverage5["pial"]["left_hemisphere"]
+    )
+    texture_right = surface.vol_to_surf(
+        fmri_img[0], fsaverage5["pial"]["right_hemisphere"]
+    )
+    image = SurfaceImage(
+        mesh={
+            "lh": fsaverage5["pial"]["left_hemisphere"],
+            "rh": fsaverage5["pial"]["right_hemisphere"],
+        },
+        data={
+            "lh": texture_left.T,
+            "rh": texture_right.T,
+        },
+    )
```

Comment:
given that fmriprep datasets can contain surface data, it would make sense for `first_level_from_bids` to return the path to the surface images if the space label corresponds to some fsaverage.